### PR TITLE
Upstream testing

### PR DESCRIPTION
**Pull request by rickducott**

### CODE REVIEW

File: projects/gloo/cli/pkg/cmd/create/upstream_interactive_test.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+			}, func() {
+				var upstream options.InputUpstream
+				err := AddUpstreamFlagsInteractive(&upstream)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(upstream.Azure.Secret).To(Equal(secretRef))
+				Expect(upstream.Azure.FunctionAppName).To(Equal(""))
+			})
+		})
+
+		It("azure interactive with custom region", func() {
```
Comment (yuval-k) [inline projects/gloo/cli/pkg/cmd/create/upstream_interactive_test.go]: by region you mean FunctionAppName?

File: projects/gloo/cli/pkg/cmd/create/upstream_interactive_test.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+				Expect(err).NotTo(HaveOccurred())
+				Expect(upstream.Aws.Secret).To(Equal(secretRef))
+				Expect(upstream.Aws.Region).To(Equal("custom-region"))
+			})
+		})
+	})
+
+	It("static upstream no hosts", func() {
+		testutil.ExpectInteractive(func(c *testutil.Console) {
+			c.ExpectString(upstreamPrompt)
```
[the remark this answers is not in the document]
Comment (rickducott) [reply projects/gloo/cli/pkg/cmd/create/upstream_interactive_test.go]: nice! did not realize i could do this

[the remark this answers is not in the document]
Comment (yuval-k) [reply projects/gloo/cli/pkg/cmd/create/upstream_interactive_test.go]: yes, i imagine the pressdown feature more for like 'yes'\'no' questions

File: projects/gloo/cli/pkg/cmd/create/upstream_test.go [hunk 3]
```diff
@@ -17,28 +17,176 @@ var _ = Describe("Upstream", func() {
 		helpers.UseMemoryClients()
 	})
 
-	It("should create static upstream", func() {
-		err := testutils.Glooctl("create upstream static jsonplaceholder-80 --static-hosts jsonplaceholder.typicode.com:80")
-		Expect(err).NotTo(HaveOccurred())
+	const (
+		nameError = "name must be specified in flag (--name) or via first arg"
```
Comment (rickducott) [inline projects/gloo/cli/pkg/cmd/create/upstream_test.go]: In another PR, I will extract this to a constant in `argutils.metadata.go`

**Pull request merged**